gcc/config/riscv/riscv.c

```diff
@@ -2420,7 +2427,7 @@ riscv_fpr_return_fields (const_tree valtype, tree fields[2])
       if (!SCALAR_FLOAT_TYPE_P (TREE_TYPE (field)))
 	return 0;
 
-      if (i == ARRAY_SIZE (fields))
+      if (i == 2)
```

Comment:
gcc 6.1 will issue warring for that, and make --enable-werror-always build fail for gcc...

```
riscv.c:2488:31: error: 'sizeof' on array function parameter 'fields' will return size of 'tree_node**' [-Werror=sizeof-array-argument]
```